src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClient.cs

```diff
@@ -570,17 +799,23 @@ private bool ContactPointMessages(object message)
 
     private void SendGetContacts()
     {
-        var sendTo = _contacts.Length switch
+        var sendTo = _contacts.Count switch
         {
             0 => _initialContactsSelections,
-            1 => _initialContactsSelections.Union(_contacts).ToArray(),
+            1 => _initialContactsSelections.Union(_contacts),
             _ => _contacts
         };
 
+        if (sendTo.Count == 0)
+        {
+            _log.Debug("Could not send GetContacts, _initialContactsSelections and _contacts are empty.");
+            return;
+        }
+
         if (_log.IsDebugEnabled)
-            _log.Debug("Sending GetContacts to [{0}]", string.Join(", ", sendTo.AsEnumerable()));
+            _log.Debug("Sending GetContacts to [{0}]", string.Join(", ", sendTo.Select(c => c.Path).AsEnumerable()));
 
-        sendTo.ForEach(c => c.Tell(ClusterReceptionist.GetContacts.Instance));
+        sendTo.ForEach(c => c.Selection.Tell(ClusterReceptionist.GetContacts.Instance));
     }
```

Comment:
This currently didn't work

Reply:
In the original code, `_initialContactsSelections` and `_contacts` are arrays or `ActorSelection`. I just merged the `ActorSelection` and `ActorPath` so we can remove dead nodes efficiently.

Reply:
What does the debug output look like here?

Reply:
Fixed, not an issue